sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -791,7 +791,7 @@ object SparkSession {
       // Get the session from current thread's active session.
       var session = activeThreadSession.get()
       if ((session ne null) && !session.sparkContext.isStopped) {
-        options.foreach { case (k, v) => session.conf.set(k, v) }
+        options.foreach { case (k, v) => session.sessionState.conf.setConfString(k, v) }
```

Comment:
yea, the goal is to forbid users to set/unset global sql conf via public API, but here we do need to set global sql conf with sparkConf, so we should use internal API here instead.